src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -5339,6 +5339,66 @@ void MacroAssembler::safepoint_isb() {
 #endif
 }
 
+Assembler::SIMD_Arrangement elemBytes_to_Arrangement(int esize, bool isQ) {
+  switch(esize) {
+    case 1:
+      return isQ ? Assembler::T16B : Assembler::T8B;
+    case 2:
+      return isQ ? Assembler::T8H : Assembler::T4H;
+    case 4:
```

Comment:
This should be

`
      return isQ ? T8H : T4H;
`
but isn't this just a table lookup?

```
if (in_range)
    return elemBytes_to_Arrangement_table[esize][isQ]
```
Why not?

Reply:
Thanks. The table is done.